store/store.go

```diff
@@ -40,6 +40,13 @@ func New(kv KVStore) Store {
 	}
 }
 
+// SetHeight sets the height saved in the Store if it is higher than the existing height
+func (s *DefaultStore) SetHeight(height uint64) {
+	if height > atomic.LoadUint64(&s.height) {
+		atomic.StoreUint64(&s.height, height)
+	}
```

Comment:
I think your usage of atomics is unsafe (as `s.height` might theoretically be changed between `LoadUint64` and `StoreUint64`, which probably never gonna happen in this particular case). 

```suggestion
	storeHeight := atomic.LoadUint64(&s.height)
	if height > storeHeight {
		_ = atomic.CompareAndSwapUint64(&s.height, storeHeight, height)
	}
```